### PR TITLE
[Modules] Uplifted the SQL Server - Vulnerability Assessment child module to support Storage RBAC

### PR DESCRIPTION
**Pull request by ahmadabdalla**
# Description

- Aligned SQL Vulnerability Assessment module to SQL MI https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/servers/vulnerabilityassessments?pivots=deployment-language-bicep
- Documentation Link: https://learn.microsoft.com/en-us/azure/azure-sql/database/sql-database-vulnerability-assessment-storage?view=azuresql#prerequisites
- Added new test case for `vulnAss` similar to SQL MI
- Updated SQL MI module as well to align

-->

### Pipeline references

| Pipeline |
| - |
|[Sql - Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.servers.yml) |
| [Sql - ManagedInstances](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.managedinstances.yml) - Static tests pass. The deployment will most likely fail as we have a stale `min` deployment. Should confirm functionality for the `VlnAss` test case |

image

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/sql/server/vulnerability-assessment/main.bicep [hunk 1]
```diff
@@ -17,12 +17,20 @@ param recurringScansEmailSubscriptionAdmins bool = false
...
+@description('Optional. Use Access Key to access the storage account. The storage account cannot be behind a firewall or virtual network. If an access key is not used, the SQL Server system assigned managed identity must be assigned the Storage Blob Data Contributor role on the storage account.')
+param useStorageAccountAccessKey bool = false
+
+@description('Optional. Create the Storage Blob Data Contributor role assignment on the storage account. Note, the role assignment must not already exist on the storage account.')
+param createStorageRoleAssignment bool = true
 
 @description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
 param enableDefaultTelemetry bool = true
 
+var splitStorageAccountResourceId = split(storageAccountResourceId, '/')
```
Comment (AlexanderSehr) [inline modules/sql/server/vulnerability-assessment/main.bicep:32]: Seems a bit odd to me if we keep this var around whole only using it in one location. Wouldn't it be easier to just have it's value in the location where it is used? I.e., last(split(storageAccountResourceId, '/'))

File: modules/sql/server/vulnerability-assessment/.bicep/nested_storageRoleAssignment.bicep [hunk 2]
```diff
@@ -0,0 +1,17 @@
+param storageAccountName string
```
Comment (AlexanderSehr) [inline modules/sql/server/vulnerability-assessment/.bicep/nested_storageRoleAssignment.bicep:1]: Please rename folfer to "modules" as we just changed it as part of the Rbac PRs

File: modules/sql/server/vulnerability-assessment/main.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply modules/sql/server/vulnerability-assessment/main.bicep:32]: don't mind doing that. We should then align the SQL MI one too. Agree? I can do it using this PR as well

File: modules/sql/server/vulnerability-assessment/.bicep/nested_storageRoleAssignment.bicep (hunk 2 above)
Comment (ahmadabdalla) [reply modules/sql/server/vulnerability-assessment/.bicep/nested_storageRoleAssignment.bicep:1]: don't mind doing that. We should then align the SQL MI one too. Agree? I can do it using this PR as well

Comment (ahmadabdalla) [reply modules/sql/server/vulnerability-assessment/.bicep/nested_storageRoleAssignment.bicep]: @AlexanderSehr updated both modules. SQL Serv ran successfully. SQL MI static tests passed (module dep in progress)

File: modules/sql/server/vulnerability-assessment/main.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply modules/sql/server/vulnerability-assessment/main.bicep]: @AlexanderSehr updated both modules. SQL Serv ran successfully. SQL MI static tests passed (module dep in progress)

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**